lib/thor/parser/arguments.rb

```diff
@@ -30,7 +30,12 @@ def initialize(arguments = [])
 
       arguments.each do |argument|
         if !argument.default.nil?
-          @assigns[argument.human_name] = argument.default
+          begin
+            @assigns[argument.human_name] = argument.default.dup
+          rescue TypeError  # Compatibility shim for un-dup-able Fixnum in Ruby < 2.4
+            @assigns[argument.human_name] = argument.default
+            next
```

Comment:
Before Ruby 2.4, attempting to dup a Fixnum results in a TypeError. From what I can see, that's the only accepted type for an option that could possibly throw, so for the sake of backward's compatibility I've had to add this in

Reply:
So we need to call `next`?

Reply:
Ah, you're right we don't. I'll remove